### DISCUSSION TITLE
What's the difference between GirCore-Adw and GirCore-Gtk-4.0?

### DISCUSSION THREAD

**Discussion by HubKing** [gircore/gir.core#896] [Q&A] [answered]
I am interested in creating GTK 4.0 applications using C#. I saw an example project (Denaro), and it was using GirCore-Adw. But When I used Adw.ApplicationWindow, it had no decoration (like the title bar) but just a rectangle area. I saw another sample code and switched it to Gtk.ApplicationWindow, and then I had a regular decorated window.

It seems that the API's are duplicates. That is, the same classes and methods exists for both Adw and Gtk. What's the difference? Why would one use Adw over Gtk? Is Adw a better new one than GTK 4.0?

**Comment by fsobolev** [comment, marked as answer]
Adw is libadwaita, which is extension to GTK. It uses modern styling and gives additional widgets on top of what GTK offers.
You can find libadwaita docs here: https://gnome.pages.gitlab.gnome.org/libadwaita/doc/1-latest/
And here's GTK: https://docs.gtk.org/gtk4/

As you can find in libadwiata docs, indeed Adw.Window doesn't have title bar, it's expected to manually add window content with Adw.HeaderBar in it (or other custom decorations, but a headerbar works for most cases).

> Why would one use Adw over Gtk? Is Adw a better new one than GTK 4.0?

I would say like that: GTK is general purpose toolkit, while libadwaita is extension to it for GNOME apps. If you want to create apps that follow modern GNOME look and trends, you definitely want to use libadwaita.
